src/ucs/sys/string.h

```diff
@@ -26,12 +27,12 @@ BEGIN_C_DECLS
 #define UCS_VALUE_AUTO_STR "auto"
 
 /* the numeric value of "infinity" */
-#define UCS_MEMUNITS_INF    SIZE_MAX
-#define UCS_ULUNITS_INF     SIZE_MAX
+#define UCS_MEMUNITS_INF    UCS_MASK_SAFE(64)
+#define UCS_ULUNITS_INF     UCS_MASK_SAFE(64)
 
 /* value which specifies "auto" for a numeric variable */
-#define UCS_MEMUNITS_AUTO   (SIZE_MAX - 1)
-#define UCS_ULUNITS_AUTO    (SIZE_MAX - 1)
+#define UCS_MEMUNITS_AUTO   (UCS_MASK_SAFE(64) - 1)
```

Comment:
i think that in this context, better do (size_t)-1 (size_t)-2 if possible
because mask does not make sense here, it's not a bitmask like address pack flags or tag mask

Reply:
ok, done